onnxruntime/core/providers/openvino/backend_utils.cc

```diff
@@ -141,82 +154,86 @@ void SetIODefs(const ONNX_NAMESPACE::ModelProto& model_proto,
-      for (size_t j = 0; j < num_dims; j++) {
-        output_shape[j] = static_cast<int64_t>(graph_output_dims[j]);
-      }
-      auto it = output_names.find(output_info_iter->first);
-      if (it == output_names.end()) {
-        ORT_THROW(log_tag + "Output names mismatch between OpenVINO and ONNX");
-      }
-      int index = it->second;
+                 std::string output_name,
+                 std::unordered_map<std::string, int> output_names) {
 
-      output_tensors.push_back(ort.KernelContext_GetOutput(context, index, output_shape, num_dims));
-      delete output_shape;
-    }
+  OrtValue* output_tensor;
+
+  auto graph_output_blob = infer_request->GetBlob(output_name);
+  auto graph_output_dims = graph_output_blob->getTensorDesc().getDims();
+  if (batch_size > 1) {
+    // Add the batch size as dim 0.
+    graph_output_dims.insert(graph_output_dims.begin(), batch_size);
   }
-#if defined(OPENVINO_2020_4)
-  for(auto item : const_output_map){
-    auto it = output_names.find(item.first);
-    if(it == output_names.end()) {
-      ORT_THROW(log_tag + "Output names mismatch between OpenVINO and ONNX");
-    }
-    int index = it->second;
-    auto node = item.second;
-    auto shape = node->get_shape();
-
-    size_t num_dims = shape.size();
-    auto output_shape = new int64_t[num_dims];
-    for(size_t j = 0; j < num_dims; j++){
-      output_shape[j] = static_cast<int64_t>(shape[j]);
-    }
+  size_t num_dims = graph_output_dims.size();
+  auto output_shape = new int64_t[num_dims];
+  for (size_t j = 0; j < num_dims; j++) {
+    output_shape[j] = static_cast<int64_t>(graph_output_dims[j]);
+  }
+  auto it = output_names.find(output_name);
+  if (it == output_names.end()) {
+    ORT_THROW(log_tag + "Output names mismatch between OpenVINO and ONNX");
+  }
+  int index = it->second;
+
+  output_tensor = ort.KernelContext_GetOutput(context, index, output_shape, num_dims);
+  delete output_shape;
 
-    output_tensors.push_back(ort.KernelContext_GetOutput(context, index, output_shape, num_dims));
-    delete output_shape;
+  return output_tensor;
+}
+
+#if (defined OPENVINO_2020_4) || (defined OPENVINO_2021_1)
+OrtValue*
+GetOutputTensor(Ort::CustomOpApi& ort, OrtKernelContext* context,
+                 std::string output_name,
+                 std::unordered_map<std::string, int> output_names,
+                 std::shared_ptr<ngraph::Node> node){
+
+  OrtValue* output_tensor;
+  auto it = output_names.find(output_name);
+  if (it == output_names.end()) {
+    ORT_THROW(log_tag + "Output names mismatch between OpenVINO and ONNX");
   }
-#endif
-  return output_tensors;
+  int index = it->second;
+  auto shape = node->get_shape();
+
+  size_t num_dims = shape.size();
+  auto output_shape = new int64_t[num_dims];
+  for(size_t j = 0; j < num_dims; j++){
+    output_shape[j] = static_cast<int64_t>(shape[j]);
+  }
+  output_tensor = ort.KernelContext_GetOutput(context, index, output_shape, num_dims);
+  delete output_shape;
```

Comment:
>delete output_shape; [](start = 2, length = 20)

should use delete[] instead

Reply:
Fixed

Reply:
line 206 was fixed, still another one at line 232

---
In reply to: [502870083](https://github.com/microsoft/onnxruntime/pull/5431#discussion_r502870083) [](ancestors = 502870083)

Reply:
oops my bad. fixed.